tools/windows/DatadogAgentInstaller/CustomActions/Rollback/FileStorageRollbackData.cs

```diff
@@ -0,0 +1,57 @@
+using System;
+using System.Security.AccessControl;
+using Datadog.CustomActions.Interfaces;
+using Newtonsoft.Json;
+
+namespace Datadog.CustomActions.Rollback
+{
+    class FileStorageRollbackData : IRollbackAction
+    {
+        [JsonProperty("FilePath")] private string _filePath;
+        [JsonProperty("Content")] private string _content;
+
+        [JsonConstructor]
+        public FileStorageRollbackData()
+        {
+        }
+
+        public FileStorageRollbackData(string filePath)
+        {
+            _filePath = filePath;
+
+            // check that file path exists
+            if (!System.IO.File.Exists(filePath))
+            {
+                _content = null;
+            }
+            else
+            {
+                // read the content and base64 encode it
+                var fileContent = System.IO.File.ReadAllBytes(filePath);
+                _content = Convert.ToBase64String(fileContent);
+            }
+
+        }
+
+        /// <summary>
+        /// Write fileContent to the @FilePath for restore purposes.
+        /// </summary>
+        /// <remarks>
+        /// Files with no content will still be created.
+        /// </remarks>
+        public void Restore(ISession session, IFileSystemServices _, IServiceController __)
+        {
+            // restore the file content
+            if (_content == null)
+            {
+                session.Log($"File {_filePath} has no content");
+                // create an empty file
+                System.IO.File.Create(_filePath).Close();
+                return;
+            }
```

Comment:
why do we create the file here if it didn't exist at the time we set `_content = null` ? if the file exists but is empty, won't `_content` be an empty string and work okay anyway? could we add a unit test for this?